File: src/c_bindings/handle.f90
! GridTools
!
! Copyright (c) 2019, ETH Zurich
! All rights reserved.
!
! Please, refer to the LICENSE file in the root directory.
! SPDX-License-Identifier: BSD-3-Clause

module gt_handle
    implicit none
    interface
        subroutine gt_release(h) bind(c)
            use iso_c_binding
            type(c_ptr), value :: h
        end
    end interface
end

File: src/c_bindings/handle.cpp
/*
 * GridTools
 *
 * Copyright (c) 2019, ETH Zurich
 * All rights reserved.
 *
 * Please, refer to the LICENSE file in the root directory.
 * SPDX-License-Identifier: BSD-3-Clause
 */

#include <gridtools/c_bindings/handle.h>
#include <gridtools/c_bindings/handle_impl.hpp>

void gt_release(gt_handle const *obj) { delete obj; }
